File: src/main/java/net/lomeli/lomlib/core/command/CommandBaseLomLib.java
package net.lomeli.lomlib.core.command;

import net.minecraft.command.CommandBase;
import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;
import net.minecraft.server.MinecraftServer;
import net.minecraft.util.text.TextComponentString;

import net.lomeli.lomlib.util.LangUtil;

public class CommandBaseLomLib extends CommandBase {

    @Override
    public String getCommandName() {
        return null;
    }

    @Override
    public String getCommandUsage(ICommandSender sender) {
        return "command.lomlib." + getCommandName() + ".usage";
    }

    @Override
    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException {
    }

    protected void sendMessage(ICommandSender sender, String message) {
        if (sender != null)
            sender.addChatMessage(new TextComponentString(LangUtil.translate(message)));
    }
}

File: src/main/kotlin/net/lomeli/lomlib/core/command/CommandBaseLomLib.kt
package net.lomeli.lomlib.core.command

import net.lomeli.lomlib.util.LangUtil
import net.minecraft.command.CommandBase
import net.minecraft.command.CommandException
import net.minecraft.command.ICommandSender
import net.minecraft.server.MinecraftServer
import net.minecraft.util.text.TextComponentString

abstract class CommandBaseLomLib : CommandBase() {

    abstract override fun getName(): String

    override fun getUsage(sender: ICommandSender): String = "command.lomlib.$name.usage"

    @Throws(CommandException::class)
    abstract override fun execute(server: MinecraftServer, sender: ICommandSender, args: Array<String>)

    protected fun sendMessage(sender: ICommandSender?, message: String) {
        sender?.sendMessage(TextComponentString(LangUtil.translate(message)))
    }
}